### PR TITLE
chore!: Change runner labels to list[string] and apply TFLint

### PR DESCRIPTION
**Pull request by koendelaat**
Change variables `runner_labels` and `runner_extra_labels` to type: `list[string]` to better represent the content

Additionally:
- [x] Automatic determine runner labels for multi runner based on label matchers
- [x] Enable Terraform linting on all modules, including naming conventions
- [x] Enable check annotations for linting errors
- [x] Fix existing linting errors
- [ ] Apply naming conventions within objects (not supported by [TFLint](https://github.com/terraform-linters/tflint/issues/1310))

### CODE REVIEW

File: examples/multi-runner/templates/runner-configs/windows-x64.yaml [hunk 1]
```diff
@@ -7,7 +7,6 @@ delay_webhook_event: 5
 runner_config:
   runner_os: windows
   runner_architecture: x64
-  runner_extra_labels: servercore-2022
```
Comment (GuptaNavdeep1983) [inline examples/multi-runner/templates/runner-configs/windows-x64.yaml:10]: Where is this configuration coming from for multi runner?

**Comment by GuptaNavdeep1983** [comment]
I did look at the changes and they look good. It will be good to thoroughly test this change specially around the logic to filter out the extra labels from matchers.

**Comment by npalm** [comment]
@koendelaat thx, maybe a good reason as to explore the module for other variables that needs to be reanamed.

**Comment by koendelaat** [comment]
Rebased commits

**Comment by koendelaat** [comment]
@npalm Ready for review

**Comment by npalm** [comment]
close in favour of #3410

**Pull request closed without merging**